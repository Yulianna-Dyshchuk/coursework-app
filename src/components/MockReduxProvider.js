import React from "react";
import { combineReducers, createStore } from "redux";
import { Provider } from "react-redux";

const store = createStore(combineReducers({ categories: () => ([]) }));

export const MockReduxProvider = ({ children }) => {
  return <Provider store={store}>{children}</Provider>;
};
