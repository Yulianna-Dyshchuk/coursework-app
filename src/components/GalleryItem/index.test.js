import React from "react";
import { shallow } from "enzyme";
import { GalleryItem } from "./index";
import { MockReduxProvider } from "../MockReduxProvider";

describe("GalleryItem component", () => {
  it("should render correctly without props", () => {
    const component = shallow(
      <MockReduxProvider>
        <GalleryItem />
      </MockReduxProvider>
    );

    expect(component).toMatchSnapshot();
  });

  it("should render correctly with props", () => {
    const component = shallow(
      <MockReduxProvider>
        <GalleryItem
          title="TEST"
          category="all"
          shortDescription="Test"
          imageUrl="https://www.lotus-qa.com/wp-content/uploads/2020/02/testing.jpg"
        />
      </MockReduxProvider>
    );

    expect(component).toMatchSnapshot();
  });
});
