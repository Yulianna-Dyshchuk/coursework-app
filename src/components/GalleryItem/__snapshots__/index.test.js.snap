// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GalleryItem component should render correctly with props 1`] = `
<Provider
  store={
    Object {
      "@@observable": [Function],
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <GalleryItem
    category="all"
    imageUrl="https://www.lotus-qa.com/wp-content/uploads/2020/02/testing.jpg"
    shortDescription="Test"
    title="TEST"
  />
</Provider>
`;

exports[`GalleryItem component should render correctly without props 1`] = `
<Provider
  store={
    Object {
      "@@observable": [Function],
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <GalleryItem />
</Provider>
`;
